### DISCUSSION TITLE
Run direct dependencies in parallel without waiting for all all syncs to finish Airbyte<> DBT

### DISCUSSION THREAD

**Discussion by rytis-prisero** [dagster-io/dagster#18754] [Q&A] [answered]
Hello,

I have set up my first dagster flow. It works well but I cannot make to run asset dependencies directly.
This is a screenshot for the asset dependency.

![Screenshot 2023-12-15 at 11 04 14](https://github.com/dagster-io/dagster/assets/152852529/e98d9eac-0e38-4c82-b0fd-f78156b1c04b)

However this is how the job runs:
![Screenshot 2023-12-15 at 11 05 25](https://github.com/dagster-io/dagster/assets/152852529/8cf30141-8aa3-4666-bc3b-955fb98e63e9)

Is there a way that once the Airbyte google finishes, then it would directly go to the related DBT models, without waititng for other syncs to finish?

this is my job definition.
```python
import os

from dagster import Definitions, ScheduleDefinition, define_asset_job, AssetSelection
from dagster_dbt import DbtCliResource

from .assets import prisify_dbt_assets, airbyte_assets
from .constants import dbt_project_dir
from .schedules import schedules

defs = Definitions(
    assets=[prisify_dbt_assets, airbyte_assets],
    resources={
        "dbt": DbtCliResource(project_dir=os.fspath(dbt_project_dir)),
    },
    schedules=[
        ScheduleDefinition(
            job=define_asset_job("all_assets", selection='*', config={
            "execution": {
                "config": {
                    "multiprocess": {
                        "max_concurrent": 30,      # limits concurrent assets to 3
                },
            }
        }
    }), cron_schedule='0 * * * *'
        )
    ]
)

**Comment by rytis-prisero** [comment]
@sryza I have noticed that you have edited my question. Any comments or suggestion on how to achieve the desired output?

Thanks!

**Comment by rytis-prisero** [comment]
@rexledesma maybe you would have any suggestions or comments regarding my inquiry. Thanks and sorry for pinging directly.

**Comment by rexledesma** [comment, marked as answer]
This is occurring because you've scheduled all of your assets together in a single job. As a result, we wait for all the dependencies to resolve before moving on to the next set of assets. This is done to minimize the number of separate dbt invocations that are required to materialize your assets.

In this case, all these Airbyte assets are run before your dbt assets start to materialize.

If you want to change this behavior, you can separate your one job into separate schedules. Each schedule could trigger an airbyte sync and its downstream dbt assets. You could do this by using `DbtManifestAssetSelection` to select the relevant dbt sources and their downstream assets. Something like:

```python
from dagster import define_asset_job
from dagster_dbt import DbtManifestAssetSelection

define_asset_job(
    "all_assets",
    selection=DbtManifestAssetSelection(
        manifest,
        select=... # Use selection string for Airbyte source
    ).downstream() # Select assets downstream of this Airbyte asset
)
```
